app/api/server/v1/out-of-office.ts

```diff
@@ -0,0 +1,69 @@
+import { Meteor } from 'meteor/meteor';
+import { check, Match } from 'meteor/check';
+
+import { API } from '../api';
+import { settings } from '../../../settings/server';
+import { updateOutOfOffice } from '../../../out-of-office/server';
+import { OutOfOfficeUsers } from '../../../models/server';
+
+API.v1.addRoute(
+	'outOfOffice.toggle',
+	{ authRequired: true },
+	{
+		post() {
+			if (!settings.get('Out_Of_Office_Enabled')) {
+				throw new Meteor.Error('error-not-allowed', 'Out-Of-Office Not Enabled', 'outOfOffice.toggle');
+			}
+
+			check(
+				this.bodyParams,
+				Match.ObjectIncluding({
+					isEnabled: Boolean,
+					roomIds: Array,
+					customMessage: String,
+					startDate: String,
+					endDate: String,
+				}),
+			);
+
+			const { message } = (Promise as any).await(updateOutOfOffice({
```

Comment:
you can import `Promise` so you can remove this `as any`.

import from:
```ts
import { Promise } from 'meteor/promise';
```

```suggestion
			const { message } = Promise.await(updateOutOfOffice({
```

Reply:
I have added this change!